src/evm.rs

```diff
@@ -109,11 +128,90 @@ impl<S: Send + Sync + 'static + ForkSource + std::fmt::Debug> EvmNamespaceT
             }
         })
     }
+
+    fn evm_mine(&self) -> BoxFuture<Result<String>> {
+        let inner = Arc::clone(&self.node);
+        Box::pin(async move {
+            match inner.write() {
+                Ok(mut inner) => {
+                    let (keys, block, bytecodes) = {
+                        let mut storage_view = StorageView::new(&inner.fork_storage);
+                        let mut oracle_tools = OracleTools::new(&mut storage_view, HistoryEnabled);
+
+                        let bootloader_code = &inner.system_contracts.baseline_contracts;
+                        let block_context = inner.create_block_context();
+                        let block_properties =
+                            InMemoryNodeInner::<S>::create_block_properties(bootloader_code);
+
+                        let block = Block {
+                            hash: H256::random(),
+                            number: inner.current_miniblock.saturating_add(1).into(),
+                            timestamp: block_context.block_timestamp.into(),
+                            l1_batch_number: Some(block_context.block_number.into()),
+                            ..Default::default()
+                        };
+
+                        // init vm
+                        let mut vm = init_vm_inner(
+                            &mut oracle_tools,
+                            BlockContextMode::NewBlock(block_context.into(), Default::default()),
+                            &block_properties,
+                            BLOCK_GAS_LIMIT,
+                            bootloader_code,
+                            TxExecutionMode::VerifyExecute,
+                        );
+
+                        vm.execute_till_block_end(BootloaderJobType::BlockPostprocessing);
```

Comment:
I might be wrong, but provided we do not have any transactions so far (no `push_transaction_to_bootloader_memory` was called) it might be easier to short circuit the logic to merely insert the `Block` and increment the miniblock. 

I'm even unsure if we plan to support  a txpool of sorts but iirc we discussed that we'd like to stick to 1 tx per block, and that'd mean that either a `evm_mine` or `eth_sendTransaction` can occur in a block - causing it to be always empty. Thoughts?

Reply:
that was my first attempt but errors on vm invocation without rolling vm:

```
thread 'tokio-runtime-worker' panicked at 'The bootloader failed to set previous block hash. Reason: The provided block number is not correct', /home/grw/.cargo/git/checkouts/zksync-era-657998444dcd9ed9/a98e454/core/lib/vm/src/errors/tx_revert_reason.rs:96:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```